optimizer/lib/transformers/PreloadImages.js

```diff
@@ -39,23 +39,18 @@ class PreloadImages {
     const body = html.firstChildByTag('body');
     const preloadImageMap = new Map();
 
-    for (let node = body; node !== null; node = node.nextNode()) {
+    let node = body;
+    while (node !== null) {
       // We've hit the maximum number of preloads.
       if (preloadImageMap.size >= MAX_PRELOADED_IMAGES) {
         break;
       }
-      let imageUrl = this.extractImageUrl(node);
-
-      if (!imageUrl) {
-        continue;
-      }
-
-      // If srcset is used, skip preloading as we don't know which image will be used.
-      if (node.attribs.srcset) {
-        continue;
+      if (node.tagName === 'template') {
```

Comment:
I think this may break with something like this:
`
<html>
  <div>
    <h1></h1>
    <template></template>
  </div>
  <amp-img src='...'/>
</html>
`
When it gets to `template`, `node.nextSibling` will be `null`, which will then exit the loop and not process the image below it.

Reply:
done